### PR TITLE
Update angular to released typescript

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: modules/angular2/src/router/router_outlet.ts [hunk 1]
```diff
@@ -122,10 +122,11 @@ export class RouterOutlet {
 
 
   deactivate(nextInstruction: Instruction): Promise<any> {
-    return (isPresent(this.childRouter) ?
+
+    return (<Promise<any>>(isPresent(this.childRouter) ?
```
Comment (mprobst) [inline modules/angular2/src/router/router_outlet.ts]: could you extract a variable to make this more clear?

File: modules/angular2/src/http/url_search_params.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ function paramParser(rawParams: string): Map<string, List<string>> {
     var split: List<string> = StringWrapper.split(param, new RegExp('='));
     var key = ListWrapper.get(split, 0);
     var val = ListWrapper.get(split, 1);
-    var list = isPresent(map.get(key)) ? map.get(key) : [];
+    var list: List<string> = isPresent(map.get(key)) ? map.get(key) : [];
```
Comment (mprobst) [inline modules/angular2/src/http/url_search_params.ts]: don't do this, wait for MS to fix their 1.6 to support this again.

File: modules/angular2/src/util/decorators.ts [hunk 3]
```diff
@@ -49,6 +49,9 @@ export interface TypeDecorator {
    */
   <T extends Type>(type: T): T;
 
+  // Make TypeDecorator assignable to built-in ParameterDecorator type
```
Comment (mprobst) [inline modules/angular2/src/util/decorators.ts]: Needs a better comment to explain what's going on here (cannot extend the type def, why do we even need this)

File: modules/angular2/src/router/router_outlet.ts (hunk 1 above)
Comment (alexeagle) [inline modules/angular2/src/router/router_outlet.ts]: done

File: modules/angular2/src/util/decorators.ts (hunk 3 above)
Comment (alexeagle) [inline modules/angular2/src/util/decorators.ts]: done

Comment (mprobst) [inline modules/angular2/src/util/decorators.ts]: Nice. Now it just needs a full stop after the first sentence ;-)

**Comment by mprobst** [comment]
LGTM if Travis is green.

**Comment by alexeagle** [comment]
merged: 3c58878b19b015b14d2d223a063efc25ea7f1419

**Comment by pkozlowski-opensource** [comment]
Wooohooo! Thnx @alexeagle !

**Pull request closed without merging**